pkg/netpol/eval/check.go

```diff
@@ -436,3 +351,277 @@ func GetPeerExposedTCPConnections(peer Peer) *common.ConnectionSet {
+	// from the admin netpols capturing the src (if !isIngress)/ capturing the dst (if isIngress true).
+	// connections are computed considering ANPs priorities (rules of an ANP with lower priority take precedence on other ANPs rules)
+	// and rules ordering in single ANP (coming first takes precedence).
+	for _, anp := range pe.sortedAdminNetpols {
+		singleANPConns := k8s.NewPolicyConnections()
+		// collect the allowed, pass, and denied connectivity from the relevant rules into policiesConns
+		if !isIngress { // egress
+			selectsSrc, err := anp.Selects(src, false)
+			if err != nil {
+				return nil, false, err
+			}
+			// if the anp captures the src, get the relevant egress conns between src and dst
+			if selectsSrc {
+				singleANPConns, err = anp.GetEgressPolicyConns(dst)
+				if err != nil {
+					return nil, false, err
+				}
+			}
+		} else { // ingress
+			selectsDst, err := anp.Selects(dst, true)
+			if err != nil {
+				return nil, false, err
+			}
+			// if the anp captures the dst, get the relevant ingress conns (from src to dst)
+			if selectsDst {
+				singleANPConns, err = anp.GetIngressPolicyConns(src, dst)
+				if err != nil {
+					return nil, false, err
+				}
+			}
+		}
+		if !singleANPConns.IsEmpty() { // the anp is relevant (the xgress connection is captured)
+			policiesConns.CollectANPConns(singleANPConns)
+		}
+	}
+
+	if policiesConns.IsEmpty() { // conns between src and dst were not captured by the adminNetpols, to be determined by netpols/default conns
+		return nil, false, nil
+	}
+
+	return policiesConns, true, nil
+}
+
+// analyzing baseline-admin-network-policy for conns between peers (object kind == BaselineAdminNetworkPolicy):
+
+// getXgressDefaultConns returns the default connections between src and dst on the given direction (ingress/egress);
+// considering the existence of a baseline-admin-network-policy
+// if there is a BANP in the input resources, it is analyzed; if it captures xgress conns between src and dst,
+// then the captured conns are returned.
+// if there is no BANP or if the BANP does not capture connections between src and dst, then default allow-all connections is returned.
+// - note that the result may contain allowed / denied connections.
+func (pe *PolicyEngine) getXgressDefaultConns(src, dst k8s.Peer, isIngress bool) (*k8s.PolicyConnections, error) {
+	res := k8s.NewPolicyConnections()
+	if pe.baselineAdminNetpol == nil {
+		res.AllowedConns = common.MakeConnectionSet(true)
+		return res, nil
+	}
+	if isIngress { // ingress
+		selectsDst, err := pe.baselineAdminNetpol.Selects(dst, true)
+		if err != nil {
+			return nil, err
+		}
+		// if the banp selects the dst on ingress, get ingress conns
+		if selectsDst {
+			res, err = pe.baselineAdminNetpol.GetIngressPolicyConns(src, dst)
+			if err != nil {
+				return nil, err
+			}
+		} else { // egress (!isIngress)
```

Comment:
this `else` is not in the level of `if isIngress { // ingress`

Reply:
better add a test covering this if-else, that would catch this kind of bug.. also try to check tests coverage

Reply:
actually it is covered by the "tests\anp_banp_core_test"

Reply:
when fixing the bug relevant diff and connlist results of this test were changed